sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterNamespaceSetPropertiesSuiteBase.scala

```diff
@@ -0,0 +1,115 @@
+ *     - V1 In-Memory catalog:
+ *       `org.apache.spark.sql.execution.command.v1.AlterNamespaceSetPropertiesSuite`
+ *     - V1 Hive External catalog:
+ *        `org.apache.spark.sql.hive.execution.command.AlterNamespaceSetPropertiesSuite`
+ */
+trait AlterNamespaceSetPropertiesSuiteBase extends QueryTest with DDLCommandTestUtils {
+  override val command = "ALTER NAMESPACE ... SET PROPERTIES"
+
+  protected def namespace: String
+
+  protected def notFoundMsgPrefix: String
+
+  test("Namespace does not exist") {
+    val ns = "not_exist"
+    val message = intercept[AnalysisException] {
+      sql(s"ALTER DATABASE $catalog.$ns SET PROPERTIES ('d'='d')")
+    }.getMessage
+    assert(message.contains(s"$notFoundMsgPrefix '$ns' not found"))
+  }
+
+  test("basic test") {
+    val ns = s"$catalog.$namespace"
+    withNamespace(ns) {
+      sql(s"CREATE NAMESPACE $ns")
+      assert(getProperties(ns) === "")
+      sql(s"ALTER NAMESPACE $ns SET PROPERTIES ('a'='a', 'b'='b', 'c'='c')")
+      assert(getProperties(ns) === "((a,a), (b,b), (c,c))")
+      sql(s"ALTER DATABASE $ns SET PROPERTIES ('d'='d')")
+      assert(getProperties(ns) === "((a,a), (b,b), (c,c), (d,d))")
+    }
+  }
+
+  test("test with properties set while creating namespace") {
+    val ns = s"$catalog.$namespace"
+    withNamespace(ns) {
+      sql(s"CREATE NAMESPACE $ns WITH PROPERTIES ('a'='a','b'='b','c'='c')")
+      assert(getProperties(ns) === "((a,a), (b,b), (c,c))")
+      sql(s"ALTER NAMESPACE $ns SET PROPERTIES ('a'='b', 'b'='a')")
+      assert(getProperties(ns) === "((a,b), (b,a), (c,c))")
+    }
+  }
+
+  test("test reserved properties") {
+    import SupportsNamespaces._
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+    val ns = s"$catalog.$namespace"
+    withSQLConf((SQLConf.LEGACY_PROPERTY_NON_RESERVED.key, "false")) {
+      CatalogV2Util.NAMESPACE_RESERVED_PROPERTIES.filterNot(_ == PROP_COMMENT).foreach { key =>
+        withNamespace(ns) {
+          sql(s"CREATE NAMESPACE $ns")
+          val exception = intercept[ParseException] {
+            sql(s"ALTER NAMESPACE $ns SET PROPERTIES ('$key'='dummyVal')")
+          }
+          assert(exception.getMessage.contains(s"$key is a reserved namespace property"))
+        }
+      }
+    }
+    withSQLConf((SQLConf.LEGACY_PROPERTY_NON_RESERVED.key, "true")) {
+      CatalogV2Util.NAMESPACE_RESERVED_PROPERTIES.filterNot(_ == PROP_COMMENT).foreach { key =>
+        withNamespace(ns) {
+          sql(s"CREATE NAMESPACE $ns")
+          assert(getProperties(ns) === "")
+          sql(s"ALTER NAMESPACE $ns SET PROPERTIES ('$key'='foo')")
+          assert(getProperties(ns) === "", s"$key is a reserved namespace property and ignored")
+          val meta = spark.sessionState.catalogManager.catalog(catalog)
+            .asNamespaceCatalog.loadNamespaceMetadata(namespace.split('.'))
+          assert(meta.get(key) == null || !meta.get(key).contains("foo"),
```

Comment:
is this a behavior difference between v1 and v2? null vs empty string

Reply:
This was taken from `DataSourceV2SQLSuite.scala`, but further looking into this, there seems to be a difference.

In the above loop, the `key` will be either `location` or `owner`, and `loadNamespaceMetadata` returns the following:
| key | v1 catalog | v2 catalog |
|----|----------|-----------|
| location | non-null | null |
| owner | null | non-null |

I think the `null` case is interesting.
1. v2 catalog returns `null` for `location` because `CREATE NAMESPACE` doesn't create a default location if not specified, where as [v1 catalog creates a default location](https://github.com/apache/spark/blob/8f6e439068281633acefb895f8c4bd9203868c24/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L82). This is expected for v2 catalog, right?
2. v1 catalog returns `null` for `owner` since the following doesn't set `owner` to metadata: https://github.com/apache/spark/blob/8f6e439068281633acefb895f8c4bd9203868c24/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala#L347-L355. Do you know if this was intentional?

Reply:
For the location, I think it's OK. The catalog implementation should decide the default location (or even no location if the source is not file-based). We should accept this difference.

For the owner, it seems a bug that `V2SessionCatalog` does not propagate the owner field.

Reply:
For the owner, the difference comes because v2 command always adds the owner when the namespace is created:
https://github.com/apache/spark/blob/7b000116575cdc8e42653a694d9ad5273346fa61/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CreateNamespaceExec.scala#L45-L47
, whereas for v1 command doesn't add the owner when the database is created.

Instead, for v1 Hive catalog, the user property is inserted when the database is retrieved:
https://github.com/apache/spark/blob/7b000116575cdc8e42653a694d9ad5273346fa61/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L370-L373

Meanwhile, the v1 in-memory catalog implementation doesn't add the owner when the database is retrieved, so we see `null` owner above. (and the owner is a part of the property, so updating `V2SessionCatalog` doesn't really address the issue).

One thing we can do is to update the v1 in-memory catalog to add the owner when the database is created or retrieved, but it is still not consistent since adding the owner is a responsibility of the command in v2, but a responsibility of the catalog in v1. Any thoughts?

Reply:
It may be too risky to change the v1 behavior now (Hive metastore fills the owner field). Let's just update the v1 in-memory catalog to fill the owner field as well.

@yaooqinn which one do you think should set the owner field? Spark or the underlying catalog?

Reply:
for v1 and v2 database and table creation, we both respect the `sparkUser` now

Reply:
for alter properties and if it's not an explicitly ower change, we shall respect the catalog settings

Reply:
I added an owner while creating a database in `InMemoryCatalog`, explicitly added a location while creating a namespace in the test (to handle the difference for v2 catalog), and removed the `meta.get(key) == null` check.